src/python/pants/backend/python/rules/python_test_runner.py

```diff
@@ -31,8 +31,14 @@ def run_python_test(transitive_hydrated_target):
   target_root = transitive_hydrated_target.root
 
   # TODO: Inject versions and digests here through some option, rather than hard-coding it.
-  pex_snapshot = yield Get(Snapshot, UrlToFetch("https://github.com/pantsbuild/pex/releases/download/v1.5.2/pex27",
-                                                Digest('8053a79a5e9c2e6e9ace3999666c9df910d6289555853210c1bbbfa799c3ecda', 1757011)))
+  pex_name, digest = {
+    (2, 7): ("pex27", Digest('0ecbf48e3e240a413189194a9f829aec10446705c84db310affe36e23e741dbc', 1812737)),
+    (3, 6): ("pex36", Digest('ba865e7ce7a840070d58b7ba24e7a67aff058435cfa34202abdd878e7b5d351d', 1812158)),
+    (3, 7): ("pex37", Digest('51bf8e84d5290fe5ff43d45be78d58eaf88cf2a5e995101c8ff9e6a73a73343d', 1813189))
+  }[sys.version_info[0:2]]
```

Comment:
This is not a friendly key error to propagate. Consider a `get` coupled with `None` check and meaningful raise.

Reply:
Good suggestion. Let me know if there's a better exception to raise than `OSError`!